### PR TITLE
RUBY-1999 aws request construction and signature calculation


### PR DESCRIPTION
**Pull request by p-mongo**
https://jira.mongodb.com/browse/RUBY-1999

### CODE REVIEW

File: spec/integration/aws_auth_request_spec.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
+require 'lite_spec_helper'
+require 'net/http'
+
+describe Mongo::Auth::Aws::Request do
+  require_aws_auth
+
+  let(:access_key_id) { ENV.fetch('MONGO_RUBY_DRIVER_AWS_AUTH_ACCESS_KEY_ID') }
+  let(:secret_access_key) { ENV.fetch('MONGO_RUBY_DRIVER_AWS_AUTH_SECRET_ACCESS_KEY') }
+  let(:session_token) { ENV['MONGO_RUBY_DRIVER_AWS_AUTH_SESSION_TOKEN'] }
```
Comment (egiurleo) [inline spec/integration/aws_auth_request_spec.rb]: Should we put these in the spec config like the FLE environment variables?

Comment (p-mongo) [reply spec/integration/aws_auth_request_spec.rb]: There are quite a few more env vars going with the aws auth testing, let's revisit this question once all of the tests land?

Comment (egiurleo) [reply spec/integration/aws_auth_request_spec.rb:9]: Sure thing.

File: spec/NOTES.aws-auth.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+- [Signing AWS requests overview](https://docs.aws.amazon.com/general/latest/gr/sigv4_signing.html)
+- [Creating canonical request](https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html)
+- [Creating string to sign](https://docs.aws.amazon.com/general/latest/gr/sigv4-create-string-to-sign.html)
+- [Calculating signature](https://docs.aws.amazon.com/general/latest/gr/sigv4-calculate-signature.html)
+
+### Signature Debugger
+
+The most excellent [awssignature.com](http://www.awssignature.com/) was
+indispensable in debugging the actual signature calculation process.
```
Comment (ShaneHarvey) [inline spec/NOTES.aws-auth.md:184]: Neat I didn't know about awssignature.com.

**Pull request merged**